conjure-java-jackson-serialization/src/test/java/com/palantir/conjure/java/serialization/ObjectMappersTest.java

```diff
@@ -136,6 +136,26 @@ public void testMapWithDuplicateKeys() {
                         + "Key 'test' values ['bar', 'foo']");
     }
 
+    @Test
+    public void testLongDeserializationFromString() throws IOException {
+        assertThat((Long) MAPPER.readValue("\"1\"", new TypeReference<Long>() {})).isEqualTo(1L);
+    }
+
+    @Test
+    public void testLongTypeDeserializationFromString() throws IOException {
+        assertThat(MAPPER.readValue("\"1\"", Long.TYPE)).isEqualTo(1L);
+    }
+
+    @Test
+    public void testLongDeserializationFromJsonNumber() throws IOException {
+        assertThat((Long) MAPPER.readValue("1", new TypeReference<Long>() {})).isEqualTo(1L);
+    }
+
+    @Test
+    public void testLongTypeDeserializationFromJsonNumber() throws IOException {
+        assertThat(MAPPER.readValue("1", Long.TYPE)).isEqualTo(1L);
+    }
```

Comment:
Can we test with some additional cases (`Integer.MAX_VALUE + 1L`, empty, null, etc.)? QuickTheories would be nice

Reply:
Good call, updated, though I have not added the quicktheories dep in this PR.